### PR TITLE
BAH Integrating 526 Schema

### PR DESCRIPTION
**Pull request by robfelder**
## Description
This PR pulls together the [4142 Integration](https://github.com/department-of-veterans-affairs/vets-website/pull/8387/files) PR and the [Patient Acknowledgement](https://github.com/department-of-veterans-affairs/vets-website/pull/8407#pullrequestreview-155595293) PR's and ensures the payload is transformed correctly for the back end submission.

## Testing done
Manual testing. Current unit tests are broken and will be fixed in an upcoming story.

## Acceptance criteria
- [X] Ensure data is properly transformed and the form can successfully be submitted
- [X] Ensure that field names that are in the schema match the field names within vets-website
- [X] Ensure previous patient acknowledgement and 4142 integration PR's are successfully merged and work properly. 

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by imeecuison — COMMENTED** [review]
Looks great! Except for the Patient Acknowledgment piece. See my comment below.

**Review by imeecuison — DISMISSED** [review]
@JBarbour2 : has brought to my attention that when Patient Acknowledgment box UNchecked, user is still able to continue.

File: src/applications/disability-benefits/526EZ/helpers.jsx [hunk 1]
```diff
@@ -97,6 +95,33 @@ export const getReservesGuardData = formData => {
...
+/**
+* Converts the treatment date range into an array of objects from just an object
+* @param {object} treatmentDateRange object containing from/to date range
+* @returns {array} of treatmentDateRange's
+*/
+const transformDateRange = (treatmentDateRange) => {
+  return [treatmentDateRange];
+};
+
+/** Note: This function will be moved to the 526EZ helper file
```
Comment (igreulich) [inline src/applications/disability-benefits/526EZ/helpers.jsx]: Since this is in the 526EZ helper file, we can delete this.

**Review by igreulich — CHANGES_REQUESTED** [review]

**Comment by igreulich** [comment]
@imeecuison Per our discussion in today's (9/17/2018) stand-up, I am going to dismiss your rejection of the PR.

File: src/applications/disability-benefits/526EZ/helpers.jsx (hunk 1 above)
Comment (robfelder) [reply src/applications/disability-benefits/526EZ/helpers.jsx]: Fixed!

**Review by igreulich — APPROVED** [review]

**Comment by igreulich** [comment]
@robfelder Approved your changes.

**Pull request merged**